### PR TITLE
Fix ta profile page and add set cookie for csrf token

### PR DESCRIPTION
**Pull request by Tanuj22**

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -37,6 +39,10 @@ export default {
   },
   created() {
     this.updateVisibility();
+    axios.get("api/csrf-token/").then(response => {
```
Comment (ajatprabha) [inline frontend/src/App.vue]: Use httpClient here, don't create a new instance of axios.

File: src/api/urls.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
 from django.urls import path, include
 
-from api.views import AuthenticationCheckAPIView, LoginAPIView, LogoutAPIView
+from api.views import AuthenticationCheckAPIView, LoginAPIView, LogoutAPIView, CsrfTokenAPIView
 
 app_name = 'api'
 
 urlpatterns = [
+    path('token/', CsrfTokenAPIView.as_view(), name='token'),
```
Comment (ajatprabha) [inline src/api/urls.py]: This should be updated.

**Review by ajatprabha — CHANGES_REQUESTED** [review]

File: frontend/src/App.vue (hunk 1 above)
Comment (Tanuj22) [reply frontend/src/App.vue]: Done

File: src/api/urls.py (hunk 2 above)
Comment (Tanuj22) [reply src/api/urls.py]: Done

**Pull request merged**